pkg/ingester/ingester.go

```diff
@@ -356,7 +358,7 @@ type Ingester struct {
 	ingestPartitionID         int32
 	ingestPartitionLifecycler *ring.PartitionInstanceLifecycler
 
-	circuitBreaker *circuitBreaker
+	circuitBreaker *ingesterCircuitBreaker
```

Comment:
There's no need for this to be a pointer, and no need to return a pointer from `newIngesterCircuitBreaker`: if we just hold a `ingesterCircuitBreaker` struct here, we'll also handle the zero-valued case gracefully (as nil cb's inside of `ingesterCircuitBreaker` are already valid)